### ISSUE TITLE
Null pointer via TextureArray.bindImage

### ISSUE THREAD

**Issue by wivlaro** [mosra/magnum#163]
Got a null pointer when trying to access element of textureState.imageBindings

```
void AbstractTexture::bindImageInternal(const Int imageUnit, const Int level, const bool layered, const Int layer, const ImageAccess access, const ImageFormat format) {
    Implementation::TextureState& textureState = *Context::current().state().texture;
    const std::tuple<GLuint, GLint, GLboolean, GLint, GLenum> state{_id, level, layered, layer, GLenum(access)};

    /* If already bound in given texture unit, nothing to do */
    if(textureState.imageBindings[imageUnit] == state) return;

    /* Update state tracker, bind the texture to the unit */
    textureState.imageBindings[imageUnit] = state;
    glBindImageTexture(imageUnit, _id, level, layered, layer, GLenum(access), GLenum(format));
}
```

Here are screenshots of the stack traces and variables:

![screen shot 2016-07-18 at 13 25 34](https://cloud.githubusercontent.com/assets/384300/16914658/2f52cf02-4ceb-11e6-91a6-274d3aa393df.png)

It seems to be failing when trying to access item 0 of imageBindings, which has no data in it:

![screen shot 2016-07-18 at 13 28 00](https://cloud.githubusercontent.com/assets/384300/16914703/7c6ddf16-4ceb-11e6-8c3f-351065ec054e.png)

Maybe I have something wrong with my code... I'll re-check again later.

**Comment by wivlaro** [comment]
I haven't got ARB_shader_image_load_store support. That is why imageBindings isn't being initialized

**Comment by mosra** [comment]
Just to be sure we are on the same boat: texture image bindings etc. should be used only from `*Texture::bindImage()` and that should be called from your code *only* if you have image load/store supported. Calling APIs that use unsupported GL features is undefined behavior and the engine doesn't do any check internally, because that would be too much code to maintain and too much overhead. You have to do it on the app side using `Context::isExtensionSupported()` etc.

TL;DR: I think you wanted to call `Texture::bind()` instead :)

**Comment by wivlaro** [comment]
Yeah I thought I could bind an individual layer of the texture array (for
debug rendering). The documentation saying no is probably sufficient as is.

**Comment by mosra** [comment]
GL 4.3 has https://www.opengl.org/registry/specs/ARB/texture_view.txt that could be used for such thing (though it is not exposed in Magnum yet), but I'm not sure if your setup supports it if you don't even have load/store.

**Comment by mosra** [comment]
Closing as "not a bug" -- the culprit was functionality missing in the GL driver and the application code is responsible for checking for such cases.
